source/MRViewer/MRRibbonNotification.cpp

```diff
@@ -32,18 +32,108 @@ namespace MR
 
 void RibbonNotifier::pushNotification( const RibbonNotification& notification )
 {
-    if ( !notifications_.empty() && notifications_.front().notification == notification )
+    addNotification_( notifications_, notification );
+    addNotification_( notificationsHistory_, notification );
+
+    highestNotification_ = NotificationType::Count;
+    for ( const auto& nwt : notificationsHistory_ )
     {
-        notifications_.front().sameCounter++;
-        notifications_.front().timer = 0.0f;
+        if ( int( nwt.notification.type ) < int( highestNotification_ ) )
+            highestNotification_ = nwt.notification.type;
     }
-    else
+    
+    requestClosestRedraw_();
+}
+
+void RibbonNotifier::drawNotificationHistoryButton( float scaling )
+{
+    if ( !notifications_.empty() || notificationsHistory_.empty() )
```

Comment:
why check both, should check only `notificationsHistory_`

Reply:
i don't think it's necessary to show history button if we are already showing notifications
(notificationsHistory_ is empty only until first notification)